src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -51,13 +51,40 @@ public WattTimeDataSource(ILogger<WattTimeDataSource> logger, IWattTimeClient cl
     public async Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(IEnumerable<Location> locations, DateTimeOffset periodStartTime, DateTimeOffset periodEndTime)
     {
         this.Logger.LogInformation("Getting carbon intensity for locations {locations} for period {periodStartTime} to {periodEndTime}.", locations, periodStartTime, periodEndTime);
-        List<EmissionsData> result = new ();
-        foreach (var location in locations)
+        using (var activity = Activity.StartActivity())
         {
-            IEnumerable<EmissionsData> interimResult = await GetCarbonIntensityAsync(location, periodStartTime, periodEndTime);
-            result.AddRange(interimResult);
+            List<EmissionsData> result = new ();
+            foreach (var location in locations)
+            {
+                IEnumerable<EmissionsData> interimResult = await GetCarbonIntensityAsync(location, periodStartTime, periodEndTime);
+                result.AddRange(interimResult);
+            }
+            return result;
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(Location location, DateTimeOffset periodStartTime, DateTimeOffset periodEndTime)
```

Comment:
Moved from private to public to reduce extra array wrapping (aggregator function with a single location can directly call the single location data source function)